docs/troubleshooting.md

```diff
@@ -0,0 +1,69 @@
+
+## So what if I messed up while installing Azure Service Operator?
+
+If you've accidentally run through the steps out of order, or are having some trouble with the install process, check out the tips below. Thankfully, the Kubernetes eco-system provides enough flexibility that you should only need a bit of tinkering around to get the ASO running. Let's go through a scenario:
+
+Let's assume that as an end user Tats provided the wrong service principal when trying to set up ASO.
+
+**Sample scenario**: `user-A` has mistakenly used an outdated or incorrect service principal secret for `azure-service-operator` which resulted in an unsuccessful install. What can we do now?
+
+We can recognize that we reached this issue at step 4 of our install guide[here](https://operatorhub.io/operator/azure-service-operator). To remediate this, all the user needs to do is renew or refresh their service-principal credentials.
+    
+* To renew the service principal, see the command mentioned in step 4 [here](https://operatorhub.io/operator/azure-service-operator). 
+* Next, after regenerating/renewing the credentials, the user needs to re-run step 5 onwards as mentioned [here] (https://operatorhub.io/operator/azure-service-operator). However, only a few of these steps are impacted by the service principal change. For example, the step to retrieve and set the `AZURE_TENANT_ID` can be skipped.
+    
+Next, find the cloud env var for your cluster. You can find instructions [here](https://docs.microsoft.com/en-us/azure/storage/common/storage-powershell-independent-clouds#get-endpoint-using-get-azenvironment). 
+
+Once you have that, create the `azureoperatorsettings.yaml` file from step 6 mentioned [here](https://operatorhub.io/operator/azure-service-operator), and replace the placeholder values into the respective fields. Finally, run `kubectl apply -f <location of your azureoperatorsettings.yaml file>`.
+
+You should be able to see `azureoperator-controller-manager-***` pods running in your cluster. To view the pods, run `kubectl get pods -n operators` 
+
+## I have already installed `azureserviceoperator` but I'm not sure which service principal secret is in use?
+This command might be able to help you see your secret locally. 
+```kubectl get secret azureoperatorsettings -n operators -o go-template='{{range $k,$v := .data}}{{printf \"%s: \" $k}}{{if not $v}}{{$v}}{{else}}{{$v | base64decode}}{{end}}{{\"\n\"}}{{end}}'```
+[SO thread for reference](https://stackoverflow.com/questions/56909180/decoding-kubernetes-secret/58117444#58117444)
+
+## What if I see the operator namespace already exists for the cluster?
+
+If you see that the operator namespace already exists for the cluster, it's likely the result of trying to install the operator previously. It also means that you can now successfully move on to the next step. 
```

Comment:
Which step should that be?

Reply:
I can take a dig at what in this means to my mind, but I think folks with more idea can correct, essentially, there are few `yaml` config applies, so in case `cert-manager` is already applied, and we run the step again, it will show an error something like, `namespace already exist` which means user should move on to next step, also, there is a key thing which is en-user owners here: aka in case say a namespace is there but in `crashing` state then user need to make sure they restart etc.

cc: @sakthi-vetrivel